Sources/Sentry/SentryHub.m

```diff
@@ -0,0 +1,78 @@
+//
+//  SentryHub.m
+//  Sentry
+//
+//  Created by Klemens Mantzos on 11.11.19.
+//  Copyright © 2019 Sentry. All rights reserved.
+//
+
+#if __has_include(<Sentry/Sentry.h>)
+#import <Sentry/SentryHub.h>
+#import <Sentry/SentryClient.h>
+#import <Sentry/SentryBreadcrumbStore.h>
+#else
+#import "SentryHub.h"
+#import "SentryClient.h"
+#import "SentryBreadcrumbStore.h"
+#endif
+
+@interface SentryHub()
+
+@property (nonatomic, strong) SentryClient *client;
+
+@end
+
+@implementation SentryHub
+
++ (SentryHub *)defaultHub {
+    static SentryHub *_sharedInstance = nil;
+    static dispatch_once_t oncePredicate;
+    dispatch_once(&oncePredicate, ^{
+        _sharedInstance = [[self alloc] init];
+    });
+     
+    return _sharedInstance;
+}
+
+- (void)startWithOptions:(NSDictionary<NSString *,id> *)options {
```

Comment:
This should be in `startWithOptions` in `SentrySDK` this function shouldn't exist here.
There can be an `init` that takes a `client` and a `scope`.

`init` doesn't have to be in this PR but the logic should still live in `SentrySDK`

Reply:
yeah. that's right. will move this to `SentrySDK`. Still, I would like to detach `client` and `scope` from each other in a PR dedicated to this task. So my changes will move startWithOptions to `SentrySDK`, but the `SentryHub.init` will take just the client for now.

at least thats my best idea for now.